syntax = "proto3";

package catalog;

message Product {
    int32 id = 1;
    string name = 2;
}

service ProductService {
    rpc SaveProduct (SaveProductRequest) returns (Empty);
    rpc GetAllProducts (Empty) returns (AllProducts);
    rpc GetProduct (GetProductRequest) returns (GetProductReply);
    rpc DeleteProduct (DeleteProductRequest) returns (Empty);
    rpc HealthCheck (Ping) returns (Pong) {}
}

message SaveProductRequest {
    Product product = 1;
}

message DeleteProductRequest {
    int32 id = 1;
}

message GetProductRequest {
    int32 id = 1;
}

message GetProductReply {
    Product product = 1;
}

message AllProducts {
    repeated Product product = 1;
}

message Ping {
}

message Pong {
}

message Empty {
}